mitiq/zne/scaling/folding.py

```diff
@@ -334,12 +342,20 @@ def _create_weight_mask(
 
     Returns: The list of weights associated to all the gates.
     """
-    if fidelities and not all(0.0 < f <= 1.0 for f in fidelities.values()):
+    if not all(0.0 < f <= 1.0 for f in fidelities.values()):
         raise ValueError("Fidelities should be in the interval (0, 1].")
-    weights = None
-    if fidelities:
-        # Round to avoid ugly numbers like 0.09999999999999998 instead of 0.1
-        weights = {k: round(1.0 - f, 12) for k, f in fidelities.items()}
+
+    invalid_fidelities = filter(
+        lambda opname: opname.lower() not in _valid_gate_names, fidelities
+    )
+    for gate_name in invalid_fidelities:
+        warnings.warn(
+            f"You passed a fidelity for the gate '{gate_name}', but we don't "
+            "currently support this gate."
+        )
```

Comment:
We need a new test for this case.

Reply:
Good thinking, added in [0ae3a77](https://github.com/unitaryfund/mitiq/pull/1542/commits/0ae3a77e54d3a267356d7658b398963dbeae2fd0).